[google-cloud-cpp] build failure after PR 43781

### Operating system

Linux

### Compiler

GCC and Clang

### Steps to reproduce the behavior

```Shell
`git clone git@github.com:microsoft/vcpkg.git`
`./bootstrap-vcpkg.sh`
`./vcpkg install 'google-cloud-cpp[*]'`
```

### Failure logs

$ ./vcpkg install 'google-cloud-cpp[*]'
Computing installation plan...
The following packages will be built and installed:
    google-cloud-cpp[accessapproval,accesscontextmanager,advisorynotifications,aiplatform,alloydb,apigateway,apigeeconnect,apikeys,appengine,apphub,artifactregistry,asset,assuredworkloads,automl,backupdr,baremetalsolution,batch,beyondcorp,bigquery,bigquerycontrol,bigtable,billing,binaryauthorization,certificatemanager,channel,cloudbuild,cloudcontrolspartner,cloudquotas,commerce,composer,compute,confidentialcomputing,config,connectors,contactcenterinsights,container,containeranalysis,contentwarehouse,core,datacatalog,datafusion,datamigration,dataplex,dataproc,datastore,datastream,deploy,developerconnect,dialogflow-cx,dialogflow-es,discoveryengine,dlp,documentai,domains,edgecontainer,edgenetwork,essentialcontacts,eventarc,filestore,functions,gkebackup,gkeconnect,gkehub,gkemulticloud,grafeas,grpc-common,iam,iap,ids,kms,language,logging,managedidentities,managedkafka,memcache,metastore,migrationcenter,monitoring,netapp,networkconnectivity,networkmanagement,networksecurity,networkservices,notebooks,oauth2,opentelemetry,optimization,orgpolicy,osconfig,oslogin,parallelstore,policysimulator,policytroubleshooter,privateca,privilegedaccessmanager,profiler,publicca,pubsub,rapidmigrationassessment,recaptchaenterprise,recommender,redis,resourcemanager,resourcesettings,rest-common,retail,rpc,run,scheduler,secretmanager,securesourcemanager,securitycenter,securitycentermanagement,servicecontrol,servicedirectory,servicehealth,servicemanagement,serviceusage,shell,spanner,speech,sql,storage,storage-grpc,storagecontrol,storageinsights,storagetransfer,support,talent,tasks,telcoautomation,texttospeech,timeseriesinsights,tpu,trace,translate,video,videointelligence,vision,vmmigration,vmwareengine,vpcaccess,webrisk,websecurityscanner,workflows,workstations]:x64-linux@2.35.0#1
Detecting compiler hash for triplet x64-linux...
Compiler found: /usr/bin/c++
Restored 0 package(s) from ${HOME}/.cache/vcpkg/archives in 13 us. Use --debug to see more details.
Installing 1/1 google-cloud-cpp[accessapproval,accesscontextmanager,advisorynotifications,aiplatform,alloydb,apigateway,apigeeconnect,apikeys,appengine,apphub,artifactregistry,asset,assuredworkloads,automl,backupdr,baremetalsolution,batch,beyondcorp,bigquery,bigquerycontrol,bigtable,billing,binaryauthorization,certificatemanager,channel,cloudbuild,cloudcontrolspartner,cloudquotas,commerce,composer,compute,confidentialcomputing,config,connectors,contactcenterinsights,container,containeranalysis,contentwarehouse,core,datacatalog,datafusion,datamigration,dataplex,dataproc,datastore,datastream,deploy,developerconnect,dialogflow-cx,dialogflow-es,discoveryengine,dlp,documentai,domains,edgecontainer,edgenetwork,essentialcontacts,eventarc,filestore,functions,gkebackup,gkeconnect,gkehub,gkemulticloud,grafeas,grpc-common,iam,iap,ids,kms,language,logging,managedidentities,managedkafka,memcache,metastore,migrationcenter,monitoring,netapp,networkconnectivity,networkmanagement,networksecurity,networkservices,notebooks,oauth2,opentelemetry,optimization,orgpolicy,osconfig,oslogin,parallelstore,policysimulator,policytroubleshooter,privateca,privilegedaccessmanager,profiler,publicca,pubsub,rapidmigrationassessment,recaptchaenterprise,recommender,redis,resourcemanager,resourcesettings,rest-common,retail,rpc,run,scheduler,secretmanager,securesourcemanager,securitycenter,securitycentermanagement,servicecontrol,servicedirectory,servicehealth,servicemanagement,serviceusage,shell,spanner,speech,sql,storage,storage-grpc,storagecontrol,storageinsights,storagetransfer,support,talent,tasks,telcoautomation,texttospeech,timeseriesinsights,tpu,trace,translate,video,videointelligence,vision,vmmigration,vmwareengine,vpcaccess,webrisk,websecurityscanner,workflows,workstations]:x64-linux@2.35.0#1...
Building google-cloud-cpp[accessapproval,accesscontextmanager,advisorynotifications,aiplatform,alloydb,apigateway,apigeeconnect,apikeys,appengine,apphub,artifactregistry,asset,assuredworkloads,automl,backupdr,baremetalsolution,batch,beyondcorp,bigquery,bigquerycontrol,bigtable,billing,binaryauthorization,certificatemanager,channel,cloudbuild,cloudcontrolspartner,cloudquotas,commerce,composer,compute,confidentialcomputing,config,connectors,contactcenterinsights,container,containeranalysis,contentwarehouse,core,datacatalog,datafusion,datamigration,dataplex,dataproc,datastore,datastream,deploy,developerconnect,dialogflow-cx,dialogflow-es,discoveryengine,dlp,documentai,domains,edgecontainer,edgenetwork,essentialcontacts,eventarc,filestore,functions,gkebackup,gkeconnect,gkehub,gkemulticloud,grafeas,grpc-common,iam,iap,ids,kms,language,logging,managedidentities,managedkafka,memcache,metastore,migrationcenter,monitoring,netapp,networkconnectivity,networkmanagement,networksecurity,networkservices,notebooks,oauth2,opentelemetry,optimization,orgpolicy,osconfig,oslogin,parallelstore,policysimulator,policytroubleshooter,privateca,privilegedaccessmanager,profiler,publicca,pubsub,rapidmigrationassessment,recaptchaenterprise,recommender,redis,resourcemanager,resourcesettings,rest-common,retail,rpc,run,scheduler,secretmanager,securesourcemanager,securitycenter,securitycentermanagement,servicecontrol,servicedirectory,servicehealth,servicemanagement,serviceusage,shell,spanner,speech,sql,storage,storage-grpc,storagecontrol,storageinsights,storagetransfer,support,talent,tasks,telcoautomation,texttospeech,timeseriesinsights,tpu,trace,translate,video,videointelligence,vision,vmmigration,vmwareengine,vpcaccess,webrisk,websecurityscanner,workflows,workstations]:x64-linux@2.35.0#1...
-- Using cached googleapis-google-cloud-cpp-v2.35.0.tar.gz
-- Cleaning sources at /tmp/vcpkg/buildtrees/google-cloud-cpp/src/v2.35.0-09d5707ebe.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /tmp/vcpkg/downloads/googleapis-google-cloud-cpp-v2.35.0.tar.gz
-- Applying patch fix-googleapis-download.patch
-- Using source at /tmp/vcpkg/buildtrees/google-cloud-cpp/src/v2.35.0-09d5707ebe.clean
-- Using cached googleapis-googleapis-280725e991516d4a0f136268faf5aa6d32d21b54.tar.gz
-- Extracting source /tmp/vcpkg/downloads/googleapis-googleapis-280725e991516d4a0f136268faf5aa6d32d21b54.tar.gz
-- Using source at /tmp/vcpkg/buildtrees/google-cloud-cpp/src/6d32d21b54-d719abf35d.clean
-- Configuring x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: /tmp/vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/bin/cmake /tmp/vcpkg/buildtrees/google-cloud-cpp/src/v2.35.0-09d5707ebe.clean -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/tmp/vcpkg/packages/google-cloud-cpp_x64-linux/debug -DFETCHCONTENT_FULLY_DISCONNECTED=ON -DGOOGLE_CLOUD_CPP_ENABLE=accessapproval accesscontextmanager advisorynotifications aiplatform alloydb apigateway apigeeconnect apikeys appengine apphub artifactregistry asset assuredworkloads automl backupdr baremetalsolution batch beyondcorp bigquery bigquerycontrol bigtable billing binaryauthorization certificatemanager channel cloudbuild cloudcontrolspartner cloudquotas commerce composer compute confidentialcomputing config connectors contactcenterinsights container containeranalysis contentwarehouse datacatalog datafusion datamigration dataplex dataproc datastore datastream deploy developerconnect discoveryengine dlp documentai domains edgecontainer edgenetwork essentialcontacts eventarc filestore functions gkebackup gkeconnect gkehub gkemulticloud grafeas iam iap ids kms language logging managedidentities managedkafka memcache metastore migrationcenter monitoring netapp networkconnectivity networkmanagement networksecurity networkservices notebooks oauth2 opentelemetry optimization orgpolicy osconfig oslogin parallelstore policysimulator policytroubleshooter privateca privilegedaccessmanager profiler publicca pubsub rapidmigrationassessment recaptchaenterprise recommender redis resourcemanager resourcesettings retail rpc run scheduler secretmanager securesourcemanager securitycenter securitycentermanagement servicecontrol servicedirectory servicehealth servicemanagement serviceusage shell spanner speech sql storage storagecontrol storageinsights storagetransfer support talent tasks telcoautomation texttospeech timeseriesinsights tpu trace translate video videointelligence vision vmmigration vmwareengine vpcaccess webrisk websecurityscanner workflows workstations dialogflow_cx dialogflow_es storage_grpc -DGOOGLE_CLOUD_CPP_ENABLE_MACOS_OPENSSL_CHECK=OFF -DGOOGLE_CLOUD_CPP_ENABLE_WERROR=OFF -DGOOGLE_CLOUD_CPP_ENABLE_CCACHE=OFF -DGOOGLE_CLOUD_CPP_ENABLE_EXAMPLES=OFF -DBUILD_TESTING=OFF -DGOOGLE_CLOUD_CPP_WITH_MOCKS=OFF -DCMAKE_MAKE_PROGRAM=/tmp/vcpkg/downloads/tools/ninja/1.12.1-linux/ninja -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/tmp/vcpkg/scripts/toolchains/linux.cmake -DVCPKG_TARGET_TRIPLET=x64-linux -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/tmp/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/tmp/vcpkg -D_VCPKG_INSTALLED_DIR=/tmp/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF
    Working Directory: /tmp/vcpkg/buildtrees/google-cloud-cpp/x64-linux-dbg
    Error code: 1
    See logs for more information:
      /tmp/vcpkg/buildtrees/google-cloud-cpp/config-x64-linux-dbg-CMakeCache.txt.log
      /tmp/vcpkg/buildtrees/google-cloud-cpp/config-x64-linux-dbg-CMakeConfigureLog.yaml.log
      /tmp/vcpkg/buildtrees/google-cloud-cpp/config-x64-linux-dbg-out.log
      /tmp/vcpkg/buildtrees/google-cloud-cpp/config-x64-linux-dbg-err.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:287 (vcpkg_execute_required_process)
  ports/google-cloud-cpp/portfile.cmake:55 (vcpkg_cmake_configure)
  scripts/ports.cmake:196 (include)


error: building google-cloud-cpp:x64-linux failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle google-cloud-cpp:x64-linux: 12 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+google-cloud-cpp
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[google-cloud-cpp]+Build+error+on+x64-linux&body=Copy%20issue%20body%20from%20%2Ftmp%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md


### Additional context

I'm not certain, but I wonder if some of the changes in #43781 may have contributed to this failure.

--- a/ports/google-cloud-cpp/fix-googleapis-download.patch
+++ b/ports/google-cloud-cpp/fix-googleapis-download.patch
@@ -35,3 +35,15 @@ index 5e93f522..30132c06 100644
      set_target_properties(
          "google_cloud_cpp_${short_name}"
          PROPERTIES EXPORT_NAME google-cloud-cpp::${short_name}
+diff --git a/protos/google/cloud/compute/CMakeLists.txt b/protos/google/cloud/compute/CMakeLists.txt
+index 0a3b1ed..c054bc3 100644
+--- a/protos/google/cloud/compute/CMakeLists.txt
++++ b/protos/google/cloud/compute/CMakeLists.txt
+@@ -56,7 +56,6 @@ google_cloud_cpp_install_proto_library_headers(compute_protos)
+ # `*.proto` files. We achieve this by having this target depend on all proto
+ # libraries. It has to be defined at the top level of the project.
+ add_dependencies(google-cloud-cpp-protos compute_protos)
+-add_dependencies(compute_protos googleapis_download)
+ 
+ # Get the destination directories based on the GNU recommendations.
+ include(GNUInstallDirs)

--- a/ports/google-cloud-cpp/vcpkg.json
+++ b/ports/google-cloud-cpp/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "google-cloud-cpp",
   "version": "2.35.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "C++ Client Libraries for Google Cloud Platform APIs.",
   "homepage": "https://github.com/googleapis/google-cloud-cpp",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3230,7 +3230,7 @@
     },
     "google-cloud-cpp": {
       "baseline": "2.35.0",
-      "port-version": 1
+      "port-version": 2
     },
     "google-cloud-cpp-common": {
       "baseline": "alias",

--- a/versions/g-/google-cloud-cpp.json
+++ b/versions/g-/google-cloud-cpp.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "9092abf492bbd95bfc31a164371f2a00183ff2f0",
+      "version": "2.35.0",
+      "port-version": 2
+    },
+    {
       "git-tree": "b30c7f31e3ce20a6197872665bc9939b7d2a95a2",
       "version": "2.35.0",
       "port-version": 1

